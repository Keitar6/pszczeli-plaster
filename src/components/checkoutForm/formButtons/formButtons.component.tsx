import Button, {
  BUTTON_TYPE_CLASSES
} from "../../../components/button/button.component";
import { BaseSyntheticEvent, FC } from "react";
import { FormButtonsContainer } from "./formButtons.styles";
type FormButtonsProps = {
  submitHandler: (
    e?: BaseSyntheticEvent<object, unknown, unknown> | undefined
  ) => Promise<void>;
};

export const FormButtons: FC<FormButtonsProps> = ({ submitHandler }) => {
  return (
    <FormButtonsContainer>
      <Button
        id="submitFormButton"
        type="submit"
        buttonType={BUTTON_TYPE_CLASSES.formButton}
        onClick={(e) => {
          e.preventDefault();
          submitHandler();
        }}
      >
        Zamawiam
      </Button>
    </FormButtonsContainer>
  );
};
