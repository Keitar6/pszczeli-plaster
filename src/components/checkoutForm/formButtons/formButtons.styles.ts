import { StandardFlexContainer } from "global.styles";
import styled from "styled-components";

export const FormButtonsContainer = styled(StandardFlexContainer)`
  // border: 2px solid red;
  width: 100%;
  flex-direction: row;
  align-items: space-between;
  justify-content: space-between;
  padding: 1rem 1rem 0 1rem;
`;
