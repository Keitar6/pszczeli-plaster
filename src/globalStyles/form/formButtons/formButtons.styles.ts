import styled from "styled-components";
import { StandardFlexContainer } from "../../../global.styles";

export const FormButtonsContainer = styled(StandardFlexContainer)`
  // border: 2px solid red;
  width: 100%;
  flex-direction: row;
  align-items: space-between;w
  justify-content: space-between;
  padding: 1rem 0rem 0 0rem;
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.75));
`;
