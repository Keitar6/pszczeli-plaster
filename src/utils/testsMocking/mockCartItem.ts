import { CategoryItem } from "../../store/categories/category.types";

export const mockCartItemInArray: CategoryItem & { quantity: number } = {
  id: "ajdi",
  name: "name",
  weight: 205,
  price: 13,
  image: "yes",
  dodatki: false,
  quantity: 2
};

export const mockCartItemInArray1Quantity: CategoryItem & { quantity: number } =
  {
    id: "ajdi",
    name: "name",
    weight: 205,
    price: 13,
    image: "yes",
    dodatki: false,
    quantity: 1
  };

export const mockCartItemNotInArray: CategoryItem & { quantity: number } = {
  id: "ajdi",
  name: "name",
  weight: 205,
  price: 13,
  image: "yes",
  dodatki: false,
  quantity: 0
};

export const mockCartItems = [
  {
    id: "FajneID",
    name: "SuperName",
    weight: 15,
    price: 165,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 2
  },
];

export const mockCartItems2 = [
  {
    id: "FajneID",
    name: "SuperName",
    weight: 15,
    price: 165,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 2
  },
  {
    id: "AjneID",
    name: "UperName",
    weight: 14,
    price: 42,
    image: "ImageSciezka",
    dodatki: false,
    quantity: 1
  }
];