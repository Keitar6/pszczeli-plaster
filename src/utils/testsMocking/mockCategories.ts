export const mockCategoryItem = {
  id: "001M",
  name: "Miód lipowy",
  dodatki: false,
  image: "Miody/lipa.jpg",
  weight: 500,
  price: 25
};
export const mockCategoryItemArray = [
  {
    id: "001M",
    name: "Miód lipowy",
    dodatki: false,
    image: "Miody/lipa.jpg",
    weight: 500,
    price: 25
  },
  {
    id: "002M",
    name: "Miód rzepakowy",
    dodatki: false,
    image: "Miody/rzepak.jpg",
    weight: 500,
    price: 24
  },
  {
    id: "003M",
    name: "Miód wielokwiatowy",
    dodatki: false,
    image: "Miody/wielokwiat.jpg",
    weight: 500,
    price: 20
  }
];
export const mockCategories = [
  {
    title: "Miody",
    icon: "emojione:honey-pot",
    items: [
      {
        id: "001M",
        name: "Miód lipowy",
        dodatki: false,
        image: "Miody/lipa.jpg",
        weight: 500,
        price: 25
      },
      {
        id: "002M",
        name: "Miód rzepakowy",
        dodatki: false,
        image: "Miody/rzepak.jpg",
        weight: 500,
        price: 24
      },
      {
        id: "003M",
        name: "Miód wielokwiatowy",
        dodatki: false,
        image: "Miody/wielokwiat.jpg",
        weight: 500,
        price: 20
      }
    ]
  },
  {
    title: "Miody pitne",
    icon: "icon-park-solid:honey-one",
    items: [
      {
        id: "001MP",
        name: "Miód pitny trójniak",
        image: "MiodyPitne/rogDoMiodu.jpg",
        weight: 500,
        price: 35
      }
    ]
  },
  {
    title: "Produkty pszczele",
    icon: "noto-v1:honeybee",
    items: [
      {
        id: "001PP",
        name: "Pyłek pszczeli",
        image: "ProduktyPszczele/PylekPszczeli.jpg",
        weight: 500,
        price: 48
      }
    ]
  }
];

export const mockCategoriesMap = {
  miody: [
    {
      id: "001M",
      name: "Miód lipowy",
      dodatki: false,
      image: "Miody/lipa.jpg",
      weight: 500,
      price: 25
    },
    {
      id: "002M",
      name: "Miód rzepakowy",
      dodatki: false,
      image: "Miody/rzepak.jpg",
      weight: 500,
      price: 24
    },
    {
      id: "003M",
      name: "Miód wielokwiatowy",
      dodatki: false,
      image: "Miody/wielokwiat.jpg",
      weight: 500,
      price: 20
    }
  ],

  "miody pitne": [
    {
      id: "001MP",
      name: "Miód pitny trójniak",
      image: "MiodyPitne/rogDoMiodu.jpg",
      weight: 500,
      price: 35
    }
  ],

  "produkty pszczele": [
    {
      id: "001PP",
      name: "Pyłek pszczeli",
      image: "ProduktyPszczele/PylekPszczeli.jpg",
      weight: 500,
      price: 48
    }
  ]
};

export const mockAllItemsMap = [
  {
    id: "001M",
    name: "Miód lipowy",
    dodatki: false,
    image: "Miody/lipa.jpg",
    weight: 500,
    price: 25
  },
  {
    id: "002M",
    name: "Miód rzepakowy",
    dodatki: false,
    image: "Miody/rzepak.jpg",
    weight: 500,
    price: 24
  },
  {
    id: "003M",
    name: "Miód wielokwiatowy",
    dodatki: false,
    image: "Miody/wielokwiat.jpg",
    weight: 500,
    price: 20
  },

  {
    id: "001MP",
    name: "Miód pitny trójniak",
    image: "MiodyPitne/rogDoMiodu.jpg",
    weight: 500,
    price: 35
  },

  {
    id: "001PP",
    name: "Pyłek pszczeli",
    image: "ProduktyPszczele/PylekPszczeli.jpg",
    weight: 500,
    price: 48
  }
];
