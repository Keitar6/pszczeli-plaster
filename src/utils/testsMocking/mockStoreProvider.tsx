import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";

import rootReducer, { ReduxState } from "../../store/rootReducer.redux";
import { Provider } from "react-redux";
import { AppStore } from "../../store/store.redux";
import { BrowserRouter } from "react-router-dom";

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<ReduxState>;
  store?: typeof AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer as any, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  ifWithRouter = true
) {
  function Wrapper({ children }: PropsWithChildren): JSX.Element {
    if (ifWithRouter)
      return (
        <BrowserRouter>
          <Provider store={store}>{children}</Provider>
        </BrowserRouter>
      );
    else return <Provider store={store}>{children}</Provider>;
  }
  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
