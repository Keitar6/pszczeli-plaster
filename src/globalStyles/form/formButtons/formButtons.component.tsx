import { BaseSyntheticEvent, FC, PropsWithChildren } from "react";
import Button, {
  BUTTON_TYPE_CLASSES
} from "../../../components/button/button.component";
import { FormButtonsContainer } from "./formButtons.styles";
type FormButtonsProps = {
  submitHandler: (
    e?: BaseSyntheticEvent<object, unknown, unknown> | undefined
  ) => Promise<void>;
  buttonTypeProp?: BUTTON_TYPE_CLASSES;
};

export const FormButtons: FC<PropsWithChildren<FormButtonsProps>> = ({
  children,
  submitHandler,
  buttonTypeProp
}) => {
  const BTNType = !buttonTypeProp ? BUTTON_TYPE_CLASSES.formButton : buttonTypeProp;

  return (  
    <FormButtonsContainer>
      <Button
        data-testid="submit"
        id="submitFormButton"
        type="submit"
        buttonType={BTNType}
        onClick={(e) => {
          e.preventDefault();
          submitHandler();
        }}
      >
        {children}
      </Button>
    </FormButtonsContainer>
  );
};
