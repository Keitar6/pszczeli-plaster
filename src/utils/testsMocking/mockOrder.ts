import { mockCartItems } from "./mockCartItem";

export const deliveryData = {
  dobryDelivery: "bardzoDobryDelivery"
};

export const mockOrder = {
  id: "adgadg",
  time: "adgadg",
  price: 123,
  itemsBought: mockCartItems,
  deliveryData: deliveryData,
  deliveryPrice: 15
};
