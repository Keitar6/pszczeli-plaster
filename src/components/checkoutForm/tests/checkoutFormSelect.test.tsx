import { renderWithProviders } from "../../../utils/testsMocking/mockStoreProvider";
import { FormButtons } from "../formButtons/formButtons.component";

describe("checkoutFroms", () => {
  test("formButtons", () => {
    const submitHandler = jest.fn();
    const component = renderWithProviders(
      <FormButtons submitHandler={submitHandler} />
    );
    expect(component).toMatchSnapshot();
    const submitHandle = component.getByTestId("submit");
    expect(submitHandle).toBeTruthy();
    submitHandle.click();

    expect(submitHandler).toHaveBeenCalledTimes(1);
  });

  // test("CheckoutFormSelect - selectButtony i errory", () => {
  //   const register = jest.fn();
  //   const mockErrorDelivery:
  //     | FieldError
  //     | Merge<FieldError, FieldErrorsImpl<any>>
  //     | undefined = undefined;

  //   const mockErrorPay:
  //     | FieldError
  //     | Merge<FieldError, FieldErrorsImpl<any>>
  //     | undefined = undefined;

  //   const component = ComponentsRenderer.create(
  //     <CheckoutFormSelect
  //       idDelivery="AJDIDel"
  //       idPayment="AJDIPay"
  //       register={register}
  //       errorPayment={mockErrorPay}
  //       errorDelivery={mockErrorDelivery}
  //     />
  //   ).toJSON();
  //   expect(component).toMatchSnapshot();
  // });
});
