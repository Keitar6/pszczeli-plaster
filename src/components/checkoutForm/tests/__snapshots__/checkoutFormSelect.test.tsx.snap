// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkoutFroms formButtons 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  min-width: 10rem;
  width: auto;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 1rem 3rem;
  font-size: 1 rem;
  background-color: #faeff0;
  font-family: "Open Sans Condensed";
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:hover {
  background-color: #023047;
  color: #FFB703;
}

.c2:focus {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

.c3 {
  width: 100%;
  border: 2px solid;
  border-radius: 0.5rem;
  border-image-slice: 1;
  background: #faeff0;
  border-image-source: linear-gradient( to left, #023047, #FFB703, #023047 );
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1rem 0rem 0 0rem;
}

<body>
    <div>
      <div
        class="c0 c1"
      >
        <button
          class="c2 c3"
          data-testid="submit"
          id="submitFormButton"
          type="submit"
        >
          Zamawiam
        </button>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  min-width: 10rem;
  width: auto;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 1rem 3rem;
  font-size: 1 rem;
  background-color: #faeff0;
  font-family: "Open Sans Condensed";
  font-weight: bolder;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:hover {
  background-color: #023047;
  color: #FFB703;
}

.c2:focus {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
  -webkit-transition: all 0.6s ease;
  transition: all 0.6s ease;
}

.c3 {
  width: 100%;
  border: 2px solid;
  border-radius: 0.5rem;
  border-image-slice: 1;
  background: #faeff0;
  border-image-source: linear-gradient( to left, #023047, #FFB703, #023047 );
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1rem 0rem 0 0rem;
}

<div>
    <div
      class="c0 c1"
    >
      <button
        class="c2 c3"
        data-testid="submit"
        id="submitFormButton"
        type="submit"
      >
        Zamawiam
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
